### PR TITLE
Update the docs w/ JS configuration info

### LINKED ISSUE #2286: Document tracing on JavaScript
Should be similar to the JVM tracing docs and in the same section. Basically we just need to know how to configure it, and users also need to be aware that it's there.

### PR DESCRIPTION
**Pull request by armanbilge**
Lots of reorganizing here, let me know what you think.

### CODE REVIEW

File: docs/scaling-and-tuning/io-runtime-config.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+| `cats.effect.logNonDaemonThreads.sleepIntervalMillis` <br/> N/A                                   | `Long` (`10000L`)  | Time to sleep between checking for presence of non-daemon threads.                                                                |
+| `cats.effect.cancelation.check.threshold ` <br/> `CATS_EFFECT_CANCELATION_CHECK_THRESHOLD`        | `Int` (`512`)      | Configure how often cancellation is checked. By default, every 512 iterations of the run loop.                                    |
+| `cats.effect.auto.yield.threshold.multiplier` <br/> `CATS_EFFECT_AUTO_YIELD_THRESHOLD_MULTIPLIER` | `Int` (`2`)        | `autoYieldThreshold = autoYieldThresholdMultiplier x cancelationCheckThreshold`. See [thread-model](../thread-model.md).          |
+| `cats.effect.tracing.exceptions.enhanced` br/> `CATS_EFFECT_TRACING_EXCEPTIONS_ENHANCED`          | `Boolean` (`true`) | Augment the stack traces of caught exceptions to include frames from the asynchronous stack traces. See [tracing](../tracing.md). |
```
Comment (vasilmkd) [inline docs/scaling-and-tuning/io-runtime-config.md]: `<br/>`

Comment (vasilmkd) [reply docs/scaling-and-tuning/io-runtime-config.md]: I like the table.

Comment (armanbilge) [reply docs/scaling-and-tuning/io-runtime-config.md]:
```suggestion
| `cats.effect.tracing.exceptions.enhanced` <br/> `CATS_EFFECT_TRACING_EXCEPTIONS_ENHANCED`         | `Boolean` (`true`) | Augment the stack traces of caught exceptions to include frames from the asynchronous stack traces. See [tracing](../tracing.md). |
```

File: docs/scaling-and-tuning/io-runtime-config.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+| `cats.effect.logNonDaemonThreadsOnExit` <br/> N/A                                                 | `Boolean` (`true`) | Whether or not we should check for non-daemon threads on JVM exit.                                                                |
+| `cats.effect.logNonDaemonThreads.sleepIntervalMillis` <br/> N/A                                   | `Long` (`10000L`)  | Time to sleep between checking for presence of non-daemon threads.                                                                |
+| `cats.effect.cancelation.check.threshold ` <br/> `CATS_EFFECT_CANCELATION_CHECK_THRESHOLD`        | `Int` (`512`)      | Configure how often cancellation is checked. By default, every 512 iterations of the run loop.                                    |
+| `cats.effect.auto.yield.threshold.multiplier` <br/> `CATS_EFFECT_AUTO_YIELD_THRESHOLD_MULTIPLIER` | `Int` (`2`)        | `autoYieldThreshold = autoYieldThresholdMultiplier x cancelationCheckThreshold`. See [thread-model](../thread-model.md).          |
```
Comment (vasilmkd) [inline docs/scaling-and-tuning/io-runtime-config.md]:
```suggestion
| `cats.effect.auto.yield.threshold.multiplier` <br/> `CATS_EFFECT_AUTO_YIELD_THRESHOLD_MULTIPLIER` | `Int` (`2`)        | `autoYieldThreshold = autoYieldThresholdMultiplier x cancelationCheckThreshold`. See [thread model](../thread-model.md).          |
```

**Review by vasilmkd — DISMISSED** [review]

**Comment by armanbilge** [comment]
We'll have to keep this on ice until 3.3 is out :)

**Comment by vasilmkd** [comment]
Merge to create hype and we ship it earlier.

**Review by vasilmkd — APPROVED** [review]

**Comment by armanbilge** [comment]
Ah yeah! We can merge this stuff now 🎉

**Pull request merged**